src/qibo/gates/channels.py

```diff
@@ -673,14 +673,23 @@ class ResetChannel(KrausChannel):
     def __init__(self, q, p0=0.0, p1=0.0):
         import numpy as np
 
+        if p0 < 0:
+            raise_error(ValueError, "Invalid p0 ({p0} < 0).")
+        if p1 < 0:
+            raise_error(ValueError, "Invalid p1 ({p1} < 0).")
+        if p0 + p1 > 1:
```

Comment:
This is more rebust with a precision tolerance, like `p0 + p1 > 1 + PRECISION_TOL`